libs/feature/admin/src/lib/user/edit-user-dialog.tsx

```diff
@@ -0,0 +1,263 @@
+	onClose
+}: {
+	initialUser: User;
+	username: string;
+	onClose: OnDialogCloseFn<User>;
+}) {
+	const { mutateAsync: updateUser } = trpc.admin.updateUser.useMutation();
+	const form = useForm({
+		resolver: zodResolver(userSchema),
+		defaultValues: initialUser
+	});
+
+	function onSubmit(user: User) {
+		updateUser({ username, user })
+			.then(res => {
+				showToast({
+					type: "success",
+					title: "User gespeichert!",
+					subtitle: res.displayName
+				});
+				onClose(undefined);
+			})
+			.catch(err => {
+				console.error(err);
+				showToast({
+					type: "error",
+					title: "Fehler",
+					subtitle: "User konnte nicht gespeichert werden."
+				});
+			});
+	}
+
+	return (
+		<FormProvider {...form}>
+			<form
+				className="flex flex-col justify-between overflow-hidden"
+				onSubmit={form.handleSubmit(onSubmit)}
+			>
+				<div className="absolute top-8 right-8">
+					<OpenAsJsonButton form={form} validationSchema={userSchema} />
+				</div>
+
+				<div className="grid gap-4 xl:grid-cols-[1fr_300px]">
+					<div className="rounded-lg border border-light-border p-6 xl:row-span-3">
+						<UserData />
+					</div>
+					<div className="rounded-lg border border-light-border p-4">
+						<LabeledField label="Nutzer:in löschen">
+							<ActionButtons username={username} />
+						</LabeledField>
+					</div>
+				</div>
+
+				<div className="absolute bottom-8 right-8 flex justify-end">
+					<DialogActions onClose={onClose}>
+						<button className="btn-primary" type="submit">
+							Speichern
+						</button>
+					</DialogActions>
+				</div>
+			</form>
+		</FormProvider>
+	);
+}
+
+function UserData() {
+	const { register, control, setValue, formState, watch } = useFormContext<User>();
+	const imgUrl = useWatch({ control, name: "image" });
+	const errors = formState.errors;
+	const height = window.innerHeight * 0.6;
+
+	const isValidDate = (date: Date | null | undefined) => {
+		if (!date) return false;
+		if (date === null) return false;
+		return date instanceof Date && !isNaN(date.getTime());
+	};
+
+	return (
+		<div className="overflow-y-auto overflow-x-hidden p-2" style={{ maxHeight: height }}>
+			<h2 className="mb-4 text-2xl">Daten</h2>
+			<div className="flex flex-col gap-4">
+				<LabeledField label="Name" error={errors.displayName?.message}>
+					<input className="textfield" type={"text"} {...register("name")} />
+				</LabeledField>
+				<LabeledField label="Displayname" error={errors.displayName?.message}>
+					<input className="textfield" type={"text"} {...register("displayName")} />
+				</LabeledField>
+				<LabeledField label="Email" error={errors.email?.message}>
+					<input className="textfield" type={"text"} {...register("email")} />
+				</LabeledField>
+				<LabeledField label="Role" error={errors.role?.message}>
+					<select
+						value={watch("role")}
+						onChange={v => {
+							setValue(
+								"role",
+								v.target.value as UserRole
+							); /*TODO Ask Marcel about this*/
```

Comment:
passt.  (aber kommentar natürlich entfernen)
alternativ wäre ein typeguard 


```suggestion
const isUserRole = (value: string): value is UserRole => {
  return Object.values(UserRole).includes(value);
}
// ...
const value = v.target.value;
isUserRole(value) && setValue("role", value)
```